api/src/services/listing.service.ts

```diff
@@ -1618,4 +1622,118 @@ export class ListingService implements OnModuleInit {
 
     return listing.jurisdictionId;
   }
+
+  async lotteryStatus(
+    dto: ListingLotteryStatus,
+    requestingUser: User,
+  ): Promise<Listing> {
+    const storedListing = await this.findOrThrow(
+      dto.listingId,
+      ListingViews.details,
+    );
+
+    await this.permissionService.canOrThrow(
+      requestingUser,
+      'listing',
+      permissionActions.update,
+      {
+        id: storedListing.id,
+        jurisdictionId: storedListing.jurisdictionId,
+      },
+    );
+
+    if (storedListing.status !== ListingsStatusEnum.closed) {
+      throw new BadRequestException(
+        'Lottery status cannot be changed until listing is closed.',
+      );
+    }
+
+    const isAdmin = requestingUser.userRoles?.isAdmin;
+    const currentStatus = storedListing.lotteryStatus;
+
+    // TODO: remove when all status logic has been implemented
+    let res;
+
+    switch (dto?.lotteryStatus) {
```

Comment:
Question: I think this is clean but I am curious about the team's preferences for switch statements versus just using if-logic (ie. listingApprovalNotify logic uses if). Should we aim to be consistent or is a balance of both fine?

Reply:
I think this is a good use case for a switch statement since each enum has a different action to do.